Add a `key` attribute to the `meta` tag for `og:image`

DavideSilva opened the pull request:
Why:
* Post previews were showing the Hero image of our site, that is the
  fallback image when none is provided, even when the post has a image
* This is happening due to using the `MetaHead` component inside the
  `Layout` component and also in the `[slug].tsx` file responsible for
  rendering the indiviual post pages. The posts pages use the `Layout`
  component and also call the `MetaHead` component with the post
  appropriate values. This causes the `meta` tags to be added twice to
  the rendered page. [Next.JS docs](https://nextjs.org/docs/api-reference/next/head)
  provide a way to prevent this

How:
* Adding a `key` attribute to the `og:image` meta tags to prevent
  duplicate `meta` tags

jfranciscosousa commented:
![image](https://user-images.githubusercontent.com/9119466/116550456-b490bf80-a8ee-11eb-879a-343ca983b66c.png)
this is a bit ugly no?

DavideSilva committed:
Add a `key` attribute to the `meta` tag for `og:image`

Why:
* Post previews were showing the Hero image of our site, that is the
  fallback image when none is provided, even when the post has a image
* This is happening due to using the `MetaHead` component inside the
  `Layout` component and also in the `[slug].tsx` file responsible for
  rendering the indiviual post pages. The posts pages use the `Layout`
  component and also call the `MetaHead` component with the post
  appropriate values. This causes the `meta` tags to be added twice to
  the rendered page. [Next.JS docs](https://nextjs.org/docs/api-reference/next/head)
  provide a way to prevent this

How:
* Adding a `key` attribute to the `og:image` meta tags to prevent
  duplicate `meta` tags

DavideSilva committed:
code review!
